pubsub/topics.js

```diff
@@ -105,8 +117,62 @@ function publishMessage (topicName, data, callback) {
 }
 // [END pubsub_publish_message]
 
+let publishCounterValue = 1;
+
+function getPublishCounterValue () {
+  return publishCounterValue;
+}
+
+function setPublishCounterValue (value) {
+  publishCounterValue = value;
+}
+
+// [START pubsub_publish_ordered_message]
+function publishOrderedMessage (topicName, data, callback) {
+  // Instantiates the client library
+  const pubsubClient = PubSub();
+
+  // References an existing topic, e.g. "my-topic"
+  const topic = pubsubClient.topic(topicName);
+
+  /**
+   * In Node.js, a PubSub message requires a "data" property, which can have a
+   * string or an object as its value. An optional "attributes" property can be
+   * an object of key/value pairs, where the keys and values are both strings.
+   * See https://cloud.google.com/pubsub/reference/rpc/google.pubsub.v1#google.pubsub.v1.PubsubMessage
+   *
+   * Topic#publish() takes either a single message object or an array of message
+   * objects. See https://googlecloudplatform.github.io/google-cloud-node/#/docs/pubsub/latest/pubsub/topic?method=publish
+   */
+  const message = {
+    data: data,
+
+    // Assign an order id to the message
+    attributes: {
+      orderId: '' + getPublishCounterValue()
+    }
+  };
+
+  topic.publish(message, (err, messageIds) => {
+    if (err) {
+      callback(err);
+      return;
+    }
+
+    // Update the counter value
+    setPublishCounterValue(+message.attributes.orderId + 1);
```

Comment:
That's coercion back to a number, apparently the attributes of a PubSub message must be a Map<string, string>